pkg/cluster/cloudstack.go

```diff
@@ -62,6 +62,9 @@ func cloudstackEntry() *ConfigManagerEntry {
 func processCloudStackDatacenter(c *Config, objects ObjectLookup) {
 	if c.Cluster.Spec.DatacenterRef.Kind == anywherev1.CloudStackDatacenterKind {
 		datacenter := objects.GetFromRef(c.Cluster.APIVersion, c.Cluster.Spec.DatacenterRef)
+		if datacenter == nil {
```

Comment:
We should probably move this to a separate PR, this is useful
Maybe add a few tests

Reply:
Sounds good, will move this out